cli/commands/benchmark-web-vitals.mjs

```diff
@@ -0,0 +1,216 @@
+export async function handler( opt ) {
+	if ( ! isValidTableFormat( opt.output ) ) {
+		log(
+			formats.error(
+				'The output format provided via the --output (-o) argument must be either "table" or "csv".'
+			)
+		);
+		return;
+	}
+
+	const { number: amount } = opt;
+	const results = [];
+
+	const browser = await puppeteer.launch();
+
+	for await ( const url of getURLs( opt ) ) {
+		const { completeRequests, metrics } = await benchmarkURL(
+			browser,
+			{
+				url,
+				amount,
+			}
+		);
+
+		results.push( [ url, completeRequests, metrics ] );
+	}
+
+	await browser.close();
+
+	if ( results.length === 0 ) {
+		log(
+			formats.error(
+				'You need to provide a URL to benchmark via the --url (-u) argument, or a file with multiple URLs via the --file (-f) argument.'
+			)
+		);
+	} else {
+		outputResults( opt, results );
+	}
+}
+
+async function benchmarkURL( browser, params ) {
+	/*
+	 * For now this only includes load time metrics.
+	 * In the future, additional Web Vitals like CLS, FID, and INP should be
+	 * added, however they are slightly more complex to retrieve through an
+	 * automated headless browser test.
+	 * See https://github.com/GoogleChromeLabs/wpp-research/pull/41.
+	 */
+	const metricsDefinition = {
+		FCP: {
+			listen: 'onFCP',
+			global: 'webVitalsFCP',
+			get: () => window.webVitalsFCP,
+			results: [],
+		},
+		LCP: {
+			listen: 'onLCP',
+			global: 'webVitalsLCP',
+			get: () => window.webVitalsLCP,
+			results: [],
+		},
+		TTFB: {
+			listen: 'onTTFB',
+			global: 'webVitalsTTFB',
+			get: () => window.webVitalsTTFB,
+			results: [],
+		},
+	};
+
+	let completeRequests = 0;
+	let requestNum = 0;
+
+	let scriptTag = `import { ${ Object.values( metricsDefinition ).map( ( value ) => value.listen ).join( ', ' ) } } from "https://unpkg.com/web-vitals@3?module";`;
+	Object.values( metricsDefinition ).forEach( ( value ) => {
+		scriptTag += `${ value.listen }( ( { name, delta } ) => { window.${ value.global } = name === 'CLS' ? delta * 1000 : delta; } );`;
+	} )
+
+	for ( requestNum = 0; requestNum < params.amount; requestNum++ ) {
+		const page = await browser.newPage();
+
+		// Set viewport similar to @wordpress/e2e-test-utils 'large' configuration.
+		await page.setViewport( { width: 960, height: 700 } );
+		await page.mainFrame().waitForFunction( 'window.innerWidth === 960 && window.innerHeight === 700' );
+
+		// Load the page.
+		const response = await page.goto( `${ params.url }?rnd=${ requestNum }`, { waitUntil: 'networkidle0' } );
+		await page.addScriptTag( { content: scriptTag, type: 'module' } );
+
+		if ( response.status() !== 200 ) {
+			continue;
+		}
+
+		completeRequests++;
+
+		await Promise.all(
+			Object.values( metricsDefinition ).map( async ( value ) => {
+				// Wait until global is populated.
+				await page.waitForFunction( `window.${ value.global } !== undefined` );
+
+				// Do a random click, since only that seems to trigger certain metrics like LCP.
+				await page.click( 'body' );
```

Comment:
FYI, this is [by design](https://web.dev/lcp/#:~:text=The%20browser%20will%20stop%20reporting%20new%20entries%20as%20soon%20as%20the%20user%20interacts%20with%20the%20page%20(via%20a%20tap%2C%20scroll%2C%20or%20keypress)%2C%20as%20user%20interaction%20often%20changes%20what%27s%20visible%20to%20the%20user%20(which%20is%20especially%20true%20with%20scrolling).)

As mentioned above, CLS is also only reported on visibility change.
FID requires an interaction.
INP requires an interaction AND a visibility change.